api/v1alpha1/groupsync_types.go

```diff
@@ -365,6 +365,17 @@ type AzureProvider struct {
 	// +kubebuilder:validation:Optional
 	BaseGroups []string `json:"baseGroups,omitempty"`
 
+	// Ca is a reference to a Secret or ConfigMap containing a CA certificate to communicate to Azure
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Resource Containing the CA Certificate",xDescriptors={"urn:alm:descriptor:io.kubernetes:Secret"}
+	// +kubebuilder:validation:Optional
+	Ca *ObjectRef `json:"ca,omitempty"`
```

Comment:
this is confusing to me. is this a reference to a secret or a configmap? Or are you using the object ref fields to determine that?

Reply:
because objectref can refer to objects in other namespaces so it's ususally not what you want for secrets and configmaps. you want localObjectRefs for them

Reply:
This is a object that is located on this Struct, nothing that Kubernetes provides. Inside, `ObjectRef`, there is a type field. 

This has been present on other group sync providers for some time now. This PR just adds the same functionality for Azure

Reply:
got it, thanks.